lib/improver/utilities/cube_manipulation.py

```diff
@@ -193,15 +188,377 @@ def concatenate_cubes(
+    return result
+
+
+def merge_cubes(cubes):
+    """
+    Function to merge cubes, accounting for differences in the
+    attributes, and coords.
+
+    Args:
+        cubes : Iris cubelist or Iris cube
+            Cubes to be merged.
+
+    Returns:
+        result : Iris cube
+            Merged cube.
+
+    """
+    if isinstance(cubes, iris.cube.Cube):
+        cubes = iris.cube.CubeList([cubes])
+
+    cubelist = equalise_cubes(cubes)
+
+    result = cubelist.merge_cube()
+    return result
+
+
+def equalise_cubes(cubes, merging=True):
+    """
+    Function to equalise cubes where they do not match.
+
+    Args:
+        cubes : Iris cubelist
+            List of cubes to check and equalise.
+        merging : boolean
+            Flag for whether the equalising is for merging
+            as slightly different processing is required.
+    Returns:
+        cubelist : Iris cubelist
+            List of cubes with revised cubes.
+    """
+    # Additional processing to deal with differences
+    # between cubes e.g. different cell_methods
+    # will probably need to be added but this function is
+    # mainly here to provide the structure for where
+    # these changes can be added.
+    equalise_cube_attributes(cubes)
+    _strip_var_names(cubes)
+    if merging:
+        cubelist = equalise_cube_coords(cubes)
+    else:
+        cubelist = cubes
+    return cubelist
+
+
+def equalise_cube_attributes(cubes):
+    """
+    Function to equalise attributes that do not match.
+
+    Args:
+        cubes : Iris cubelist
+            List of cubes to check the attributes and revise.
+    Warns:
+        Warning: If it does not know what to do with an unmatching
+                 attribute. Default is to delete it.
+    """
+    unmatching_attributes = compare_attributes(cubes)
+    if len(unmatching_attributes) > 0:
+        for i, cube in enumerate(cubes):
+            # Remove history.
+            if "history" in unmatching_attributes[i]:
+                cube.attributes.pop("history")
+                unmatching_attributes[i].pop("history")
+            # Normalise grid_id to ukx_standard_1
+            if "grid_id" in unmatching_attributes[i]:
+                if cube.attributes['grid_id'] in ['enukx_standard_v1',
+                                                  'ukvx_standard_v1',
+                                                  'ukx_standard_v1']:
+                    cube.attributes['grid_id'] = 'ukx_standard_v1'
+                unmatching_attributes[i].pop("grid_id")
+            # Add model_id if titles do not match.
+            if "title" in unmatching_attributes[i]:
+                model_title = cube.attributes.pop('title')
+                new_model_id_coord = build_coordinate([100*i],
+                                                      long_name='model_id',
+                                                      data_type=np.int)
+                new_model_coord = build_coordinate([model_title],
```

Comment:
This of course would mean changing a load of unit tests, and apart from my aversion, I don't think there is a good reason, so maybe ignore me.

Reply:
The value is the value set in Stage. I'm not sure replacing spaces with underscores is a good idea in case we ever need to match these strings (we would have to remember one had underscores and the other didn't). unless you feel very strongly I would rather leave as is.

Reply:
Yeah, it's fine.